File: Mesh4j/trunk/src/com/mesh4j/sync/validations/Guard.java
package com.mesh4j.sync.validations;

import com.mesh4j.sync.translator.MessageTranslator;

public class Guard {

	/// <summary>
	/// Checks an argument to ensure it isn't null.
	/// </summary>
	/// <param name="value">The argument value to check.</param>
	/// <param name="argumentName">The name of the argument.</param>
	public static void argumentNotNull(Object value, String argumentName)
	{
		if (value == null)
			throw new IllegalArgumentException(argumentName);
	}

	/// <summary>
	/// Checks a string argument to ensure it isn't null or empty.
	/// </summary>
	/// <param name="argumentValue">The argument value to check.</param>
	/// <param name="argumentName">The name of the argument.</param>
	public static void argumentNotNullOrEmptyString(String value, String argumentName) 
	{
		argumentNotNull(value, argumentName);

		if (value.length() == 0)
			throw new IllegalArgumentException(MessageTranslator.translate("ArgumentCanNotBeNullOrEmpty", argumentName));
	}
	
	public static void throwsArgumentException(String key, Object... args){
		throw new IllegalArgumentException(MessageTranslator.translate(key, args));
	}
	
	public static void throwsArgumentException(String key){
		throw new IllegalArgumentException(MessageTranslator.translate(key));
	}
	
	public static void throwsException(String key, Object... args){
		throw new MeshException(MessageTranslator.translate(key, args));
	}
	
	public static void throwsException(String key){
		throw new MeshException(MessageTranslator.translate(key));
	}
}

File: Mesh4n/trunk/Main/Guard.cs
using System;
using System.Globalization;

internal static class Guard
{
	/// <summary>
	/// Checks an argument to ensure it isn't null.
	/// </summary>
	/// <param name="value">The argument value to check.</param>
	/// <param name="argumentName">The name of the argument.</param>
	public static void ArgumentNotNull(object value, string argumentName)
	{
		if (value == null)
			throw new ArgumentNullException(argumentName);
	}

	/// <summary>
	/// Checks a string argument to ensure it isn't null or empty.
	/// </summary>
	/// <param name="argumentValue">The argument value to check.</param>
	/// <param name="argumentName">The name of the argument.</param>
	public static void ArgumentNotNullOrEmptyString(string value, string argumentName)
	{
		ArgumentNotNull(value, argumentName);

		if (value.Length == 0)
			throw new ArgumentException("Value cannot be null or an empty string.",
				argumentName);
	}

	public static void ArgumentIsInstanceOfType(object value, Type type, string argumentName)
	{
		if (!type.IsInstanceOfType(value))
		{
			throw new ArgumentException(string.Format(
				CultureInfo.InvariantCulture,
				"Value is not instance of type {0}",
				type.Name), argumentName);
		}
	}
}
